base/applications/osk/main.c

```diff
@@ -213,8 +213,20 @@ int OSK_DlgInitDialog(HWND hDlg)
         }
     }
 
-    /* Move the dialog according to the placement coordination */
-    SetWindowPos(hDlg, HWND_TOP, Globals.PosX, Globals.PosY, 0, 0, SWP_NOSIZE);
+    /*
+        Place the window (with respective placement coordinates) as topmost, above
+        every window which are not on top or are at the bottom of the Z order.
+    */
+    if (Globals.bAlwaysOnTop)
+    {
+        CheckMenuItem(GetMenu(hDlg), IDM_ON_TOP, MF_BYCOMMAND | MF_CHECKED);
+        SetWindowPos(hDlg, HWND_TOPMOST, Globals.PosX, Globals.PosY, 0, 0, SWP_NOSIZE);
```

Comment:
you also need to specify `SWP_NOMOVE | ` in the flags.

Reply:
This is actually incorrect. Using `SWP_NOMOVE` here will ignore *Globals.PosX* and *Globals.PosY* coordinate parameters placing the dialog at very top in the left. This totally defeats the purpose of placement coordination handler.

Reply:
Oh yes sorry I didn't see that you placed both functionalities (position in X-Y **and** Z-order) in the same call.